### ISSUE TITLE
Failed to build pysam wheel, throws Cython error.

### ISSUE THREAD

**Issue by jsquaredosquared** [pysam-developers/pysam#1239]
I am on Python 2.7.18 in WSL (Ubuntu 22.04.3 LTS).

Pip fails to build the pysam wheel, throwing a Cython error:

`Cython.Compiler.Errors.CompileError: pysam/libcalignedsegment.pyx:1451:21 Invalid index type 'double'`

What can I do to fix this?

**Comment by boegel** [comment]
@jsquaredosquared Do you remember how you fixed this?

Relevant XKCD: https://xkcd.com/979/

**Comment by jsquaredosquared** [comment]
Tbh I may have given up and used something else instead. I'm not even sure I found a solution. Not sure why I closed the issue on the same day 😅

Sorry I couldn't be more helpful 🥲

**Comment by jmarshall** [comment]
It means you are compiling old pysam source code that has not been adjusted to use `//` division, in conjunction with a version of Cython that defaults to treating `/` as float division. Line 1451 in that old commit is

```
p[k/2] |= seq_nt16_table[<unsigned char>s[k]] << 4 * (1 - k % 2)
```

The fix is to use current pysam, either 0.23.0 or the current **master** branch, which changes this line to `p[k//2] |= …` ensuring that this is treated as integer division.

**Comment by boegel** [comment]
Thanks a lot for the detailed info @jmarshall!
